plugins/opa/pkg/opa/opa.go

```diff
@@ -126,7 +141,29 @@ func processCheckTarget(ctx context.Context, check CustomCheck, checkInstance Cu
 		return nil, err
 	}
 	for _, obj := range list.Items {
-		newItems, err := ProcessCheckForItem(ctx, check, checkInstance, obj.Object, obj.GetName(), obj.GetKind(), obj.GetNamespace())
+		newItems, err := ProcessCheckForItem(ctx, check, checkInstance, obj.Object, obj.GetName(), obj.GetKind(), obj.GetNamespace(), &rego.InsightsInfo{InsightsContext: "Agent", Cluster: os.Getenv("FAIRWINDS_CLUSTER")})
+		if err != nil {
+			return nil, err
+		}
+		actionItems = append(actionItems, newItems...)
+	}
+	return actionItems, nil
+}
+
+func processCheckTargetV2(ctx context.Context, check CustomCheck, gk schema.GroupKind) ([]ActionItem, error) {
+	client := kube.GetKubeClient()
+	actionItems := make([]ActionItem, 0)
+	mapping, err := client.RestMapper.RESTMapping(gk)
+	if err != nil {
+		return actionItems, err
+	}
+	gvr := mapping.Resource
+	list, err := client.DynamicInterface.Resource(gvr).Namespace("").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, obj := range list.Items {
+		newItems, err := ProcessCheckForItemV2(ctx, check, obj.Object, obj.GetName(), obj.GetKind(), obj.GetNamespace(), &rego.InsightsInfo{InsightsContext: "Agent", Cluster: os.Getenv("FAIRWINDS_CLUSTER")})
```

Comment:
is `FAIRWINDS_CLUSTER` set in both admission and agent?

Reply:
Yes, and I"ve (this week) fixed cluster being available in the `insights info` function via the admission plugin.